### PR TITLE
[Documentation] Improve ASVisibility protocol documentation

### PR DESCRIPTION
**Pull request by garrettmoon**
And ASNavigationController, ASTabBarController

### CODE REVIEW

**Comment by appleguy** [comment]
@garrettmoon when convenient, this needs a rebase.

File: AsyncDisplayKit/ASVisibilityProtocols.h [hunk 1]
```diff
@@ -38,12 +50,39 @@ ASDISPLAYNODE_EXTERN_C_END
...
+ */
 - (void)visibilityDepthDidChange;
 
 @end
 
 /**
+ * ASManagesChildVisibilityDepth
+ *
+ * @discussion A protocol which should be implemented by container view controllers to allow proper
+ * propegation of visibility depth
```
Comment (hannahmbanana) [inline AsyncDisplayKit/ASVisibilityProtocols.h]: nit: propagation

File: AsyncDisplayKit/ASVisibilityProtocols.h [hunk 2]
```diff
@@ -38,12 +50,39 @@ ASDISPLAYNODE_EXTERN_C_END
...
+/**
+ * Called when visibility depth changes
+ *
+ * @discussion @c visibilityDepthDidChange is called whenever the visibility depth of the represented view controller
+ * has changed.
+ * 
+ * If implemented by a view controller container, use this method to notify child view controllers that their view
+ * depth has changed @see ASNavigationController.m
+ *
+ * If implemented on an ASViewController, use this method to clear out reduce or increase the resources that your
```
Comment (hannahmbanana) [inline AsyncDisplayKit/ASVisibilityProtocols.h]: nit: missing comma ---> clear out, reduce or increase

**Comment by hannahmbanana** [comment]
LGTM

File: AsyncDisplayKit/ASNavigationController.h [hunk 3]
```diff
@@ -16,6 +16,14 @@
 
 NS_ASSUME_NONNULL_BEGIN
 
+/**
+ * ASNavigationController
+ *
+ * @discussion ASNavigationController is a drop in replacement for UINavigationController
+ * which implements the memory efficiency improving @c ASManagesChildVisibilityDepth protocol.
```
Comment (appleguy) [inline AsyncDisplayKit/ASNavigationController.h]: "which improves memory efficiency by implementing the ... protocol"

Comment (appleguy) [inline AsyncDisplayKit/ASNavigationController.h]: Add another line with something like: You can use ASNavigationController with regular UIViewControllers, as well as ASViewControllers.  It is safe to subclass or use even where AsyncDisplayKit is not adopted.

**Pull request merged**